test/server/repositories/UrlRepository.test.ts

```diff
@@ -236,21 +356,62 @@ describe('UrlRepository', () => {
       expect(putObject).not.toHaveBeenCalled()
       expect(putObjectAcl).not.toHaveBeenCalled()
       expect(update).toHaveBeenCalledWith({ description }, expect.anything())
-      expect(scope).toHaveBeenCalledWith(['defaultScope', 'getClicks'])
+      expect(scope).toHaveBeenCalledWith([
+        'defaultScope',
+        'getClicks',
+        'getTags',
+      ])
+    })
+
+    it('should update tags on non-file links', async () => {
+      const description = 'Changes made'
+      const update = jest.fn()
+      const setTags = jest.fn()
+      const newTags = baseTags
+      scope.mockImplementation(() => urlModelMock)
+      findOne.mockResolvedValue({ ...baseUrl, isFile: false, update, setTags })
+      await expect(
+        repository.update(
+          { shortUrl: baseShortUrl },
+          { description, tags: newTags },
+        ),
+      ).resolves.toStrictEqual(expect.objectContaining({ isFile: false }))
+      expect(findOne).toHaveBeenCalledWith({
+        where: { shortUrl: baseShortUrl },
+      })
+      expect(putObject).not.toHaveBeenCalled()
+      expect(putObjectAcl).not.toHaveBeenCalled()
+      expect(tagRepository.upsertTags).toHaveBeenCalledTimes(1)
```

Comment:
perhaps a bit more detail on what `upsertTags` is called with? is it also appropriate to check on `setTags` here?

Reply:
done